### PR TITLE
Added the error level threshold feature to redpen-server

### PR DESCRIPTION
**Pull request by norm-ideal**
`redpen-server` was not working correctly with the new feature of "error threshold" that there was no way to specify to get `info` or `warn`.

With this commit they can request an error level by sending `errorLevel` within the POST form such as:

```
  var formData = {
    'document': body.getText(),
    'lang' : 'ja',
    'format' : 'json2',
    'documentParser' : 'plain',
    'config' : configs,
    'errorLevel' : 'warn'
  };
```

### CODE REVIEW

**Comment by takahi-i** [comment]
@norm-ideal Thank you very much for the valuable contribution as always. I agree that RedPen servers should support error level as a parameter. 

As for the implementation, I think RedPen should not provide setter methods to reduce the states of the objects (kind of a functional programming style) . Adding the `threshold` parameter to [public List<ValidationError> validate(Document document) ](https://github.com/redpen-cc/redpen/blob/4705c116bf33fc9b047ec0af5a5739b8c2d9ace7/redpen-core/src/main/java/cc/redpen/RedPen.java#L173) or create another interface `public List<ValidationError> validate(Document document, String threshold)` and apply it in [RedPenResource](https://github.com/redpen-cc/redpen/pull/879/files#diff-9ce4cb97d0b5be59edeef055f58e148623e02f919f92cfe3b35121413f5f814bR115) would be simple.

nits. we cannot change the default behavior changing the [default threshold](https://github.com/redpen-cc/redpen/pull/879/files#diff-6a7813af8ff5d3c85774f523ed65b1b4a64659ce2ab69cbaea4508ea8d6b087fR136).

**Comment by norm-ideal** [comment]
Thank you for your advice. I should have realised the solution, because it is much cleaner. How about like this?

**Comment by norm-ideal** [comment]
Now it works.

image

File: redpen-core/src/main/java/cc/redpen/RedPen.java [hunk 1]
```diff
@@ -162,15 +162,26 @@ public Document parse(DocumentParser parser, String content) throws RedPenExcept
...
 
     /**
-     * validate the input document collection. Note that this method call is NOT thread safe. RedPen instances need to be crated for each thread.
+     * validate the input document. Note that this method call is NOT thread safe. RedPen instances need to be crated for each thread.
      *
      * @param document document to be validated
      * @return list of validation errors
      */
     public List<ValidationError> validate(Document document) {
+        return validate(document, "error");
```
Comment (takahi-i) [inline redpen-core/src/main/java/cc/redpen/RedPen.java:171]: 👍

**Comment by takahi-i** [comment]
LGTM. Thank you for your contribution!

**Pull request merged**